pyart/graph/gridmapdisplay.py

```diff
@@ -348,31 +727,144 @@ def _find_nearest_grid_indices(self, lon, lat):
+        x_label, y_label = axis_labels
+        if x_label is None:
+            x_label = self._get_label_x()
+        if y_label is None:
+            y_label = self._get_label_z()
+        ax.set_xlabel(x_label)
+        ax.set_ylabel(y_label)
+
+    ##########################
+    # name generator methods #
+    ##########################
+
+    def generate_filename(self, field, level, ext='png'):
+        """
+        Generate a filename for a grid plot.
 
-        if vmax is None:
-            if 'valid_max' in self.grid.fields[field]:
-                vmax = self.grid.fields[field]['valid_max']
-            else:
-                vmax = 64
-        return vmin, vmax
-
-    def _parse_lon_lat(self, lon, lat):
-        """ Parse lat and lon parameters """
-        if lat is None:
-            lat = self.grid.axes['lat']['data'][0]
-        if lon is None:
-            lon = self.grid.axes['lon']['data'][0]
-        return lon, lat
-
-    def _parse_ax(self, ax):
-        """ Parse the ax parameter. """
-        if ax is None:
-            return plt.gca()
-        return ax
+        Generated filename has form:
+            grid_name_field_level_time.ext
+
+        Parameters
+        ----------
+        field : str
+            Field plotted.
+        level : int
+            Level plotted.
+        ext : str
+            Filename extension.
+
+        Returns
+        -------
+        filename : str
+            Filename suitable for saving a plot.
+
+        """
+        return common.generate_grid_filename(self.grid, field, level, ext)
+
+    def generate_grid_title(self, field, level):
+        """
+        Generate a title for a plot.
+
+        Parameters
+        ----------
+        field : str
+            Field plotted.
+        level : int
+            Verical level plotted.
+
+        Returns
+        -------
+        title : str
+            Plot title.
+
+        """
+        return common.generate_grid_title(self.grid, field, level)
+
+    def generate_longitude_slice_title(self, field, level):
+        """
+        Generate a title for a plot.
+
+        Parameters
+        ----------
+        field : str
+            Field plotted.
+        level : int
+            Longitudinal level plotted.
+
+        Returns
+        -------
+        title : str
+            Plot title.
+
+        """
+        return common.generate_longitude_slice_title(self.grid, field, level)
+
+    def generate_latitude_slice_title(self, field, level):
+        """
+        Generate a title for a plot.
+
+        Parameters
+        ----------
+        field : str
+            Field plotted.
+        level : int
+            Longitudinal level plotted.
+
+        Returns
+        -------
+        title : str
+            Plot title.
+
+        """
+        return common.generate_latitude_slice_title(self.grid, field, level)
+
+    ##########################
+    #      get methods       #
+    ##########################
+
+    def get_basemap(self):
```

Comment:
Can ARTView access the basemap using attribute access using display.basemap or getattr(display, 'basemap') so that this can be a private function?  If not, that is fine also just want to check.